teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,368 @@
+New TEPs can be checked in with a file name in the form of
+`draft-YYYYMMDD-my-title.md`.  As significant work is done on the TEP
+the authors can assign a TEP number.  No other changes should be put
+in that PR so that it can be approved quickly and minimize merge
+conflicts.  The TEP number can also be done as part of the initial
+submission if the PR is likely to be uncontested and merged quickly.
+
+### TEP Editor Role
+
+Taking a cue from the [Python PEP process][], we define the role of a
+TEP editor.  The job of an TEP editor is likely very similar to the
+[TEP editor responsibilities][] and will hopefully provide another
+opportunity for people who do not write code daily to contribute to
+Tekton.
+
+In keeping with the TEP editors which
+
+> Read the PEP to check if it is ready: sound and complete. The ideas
+> must make technical sense, even if they don't seem likely to be
+> accepted.  The title should accurately describe the content.  Edit
+> the PEP for language (spelling, grammar, sentence structure, etc.),
+> markup (for reST PEPs), code style (examples should match PEP 8 &
+> 7).
+
+TEP editors should generally not pass judgement on a TEP beyond
+editorial corrections.  TEP editors can also help inform authors about
+the process and otherwise help things move smoothly.
+
+[Python PEP process]: https://www.python.org/dev/peps/pep-0001/ [PEP editor responsibilities]: https://www.python.org/dev/peps/pep-0001/#pep-editor-responsibilities-workflow
+
+### Important Metrics
+
+It is proposed that the primary metrics which would signal the success
+or failure of the TEP process are
+
+- how many "enhancements" are tracked with a TEP
+- distribution of time a TEP spends in each state
+- TEP rejection rate
+- PRs referencing a TEP merged per week
+- number of issues open which reference a TEP
+- number of contributors who authored a TEP
+- number of contributors who authored a TEP for the first time
+- number of orphaned TEPs
+- number of retired TEPs
+- number of superseded TEPs
+
+### Prior Art
+
+The TEP process as proposed was essentially stolen from the
+[Kubernetes KEP process][], which itself is essentially stolen from the [Rust RFC
+process][] which itself seems to be very similar to the [Python PEP
+process][]
+
+[Rust RFC process]: https://github.com/rust-lang/rfcs
+[Kubernetes KEP process]: https://github.com/kubernetes/enhancements/tree/master/keps
+
+## Drawbacks
+
+Any additional process has the potential to engender resentment within
+the community. There is also a risk that the TEP process as designed
+will not sufficiently address the scaling challenges we face today. PR
+review bandwidth is already at a premium and we may find that the TEP
+process introduces an unreasonable bottleneck on our development
+velocity.
+
+It certainly can be argued that the lack of a dedicated issue/defect
+tracker beyond GitHub issues contributes to our challenges in managing
+a project as large as Kubernetes, however, given that other large
```

Comment:
```suggestion
a project as large as Tekton, however, given that other large
```

Reply:
My thought here was that it was a comparison to Kubernetes, but I think changing to Tekton makes sense too.